Cauli/CauliViewController/CauliViewController.swift

```diff
@@ -24,14 +24,25 @@ import UIKit
 
 internal class CauliViewController: UITableViewController {
 
-    typealias DisplayableFloret = Floret & Displayable
-
     private let cauli: Cauli
     private var displayableFlorets: [DisplayableFloret]
+    private var interceptableFlorets: [InterceptableFloret]
 
     init(cauli: Cauli) {
         self.cauli = cauli
-        displayableFlorets = cauli.florets.compactMap { $0 as? (DisplayableFloret) }
+        
+        var displayableFlorets: [DisplayableFloret] = []
+        var interceptableFlorets: [InterceptableFloret] = []
+        cauli.florets.forEach {
+            switch $0 {
+            case let displayableFloret as DisplayableFloret: displayableFlorets.append(displayableFloret)
+            case let interceptableFloret as InterceptableFloret: interceptableFlorets.append(interceptableFloret)
```

Comment:
This case will not happen if the floret is a displayable and an interceptable. So either you have to add a fallthrough or don't use switch cases is guess.